### ISSUE TITLE
[Package Request]: `Element.Element.Nightly`

### ISSUE THREAD

**Issue by RokeJulianLockhart** [microsoft/winget-pkgs#38444]
### **Package Requested**

- [X] I would like someone else to build the manifest.
- [ ] I would like help so I can submit the manifest.
- [X] I have performed a search and couldn't find this package.
- [X] I have checked that the installer for this package is not a standalone or portable executable or a .zip file
- [ ] I think there is a new version available and I have provided the URL.

### **Please provide the following information**

<table><tr>
	<th>Download Page URI</th>
	<td><a href="https://element.io/download#:~:text=Element%20web-,Desktop,-.">https://element.io/download#:~:text=Element%20web-,Desktop,-.</a></td>
</tr><tr>
	<th>Publisher</th>
	<td><a href="https://find-and-update.company-information.service.gov.uk/company/10873661#:~:text=in%20new%20window-,NEW%20VECTOR%20LIMITED,-Company%20number%2010873661">New Vector Limited</a></td>
</tr><tr>
	<th>Package Name</th>
	<td><code>Element.Element.Nightly</code></td>
</tr><tr>
	<th>Description</th>
	<td> <blockQuote cite="https://element.io/app-for-productivity"> All-in-one secure chat app for teams, friends and organisations. </blockQuote> </td>
</tr><tr>
	<th>Package Version</th>
	<td><a href="https://github.com/element-hq/element-desktop/releases/tag/v1.11.69-rc.0">https://github.com/element-hq/element-desktop/releases/tag/v1.11.69-rc.0</a></td>
</tr><tr>
	<th>Installer URI</th>
	<td><a href="https://packages.element.io/nightly/install/win32/x64/Element%20Nightly%20Setup.exe">https://packages.element.io/nightly/install/win32/x64/Element%20Nightly%20Setup.exe</a></td>
</tr></table>

**Comment by Trenly** [comment]
Are there any URL's to nightly versions which are retained? Having a vanity url here will cause many issues, since the package would need to be updated every day with the new SHA256

**Comment by vedantmgoyal2009** [comment]
That can be done by the automation?

**Comment by vedantmgoyal2009** [comment]
The path in the validation file is version specific but it's applied to the whole package.

**Comment by Trenly** [comment]
> The path in the validation file is version specific but it's applied to the whole package.

I disagree. If that were the case, then https://github.com/microsoft/winget-pkgs/pull/38412/files would not have been necessary to fix a Validation-Domain error

**Comment by vedantmgoyal2009** [comment]
Oh, it looks like they have made changes in the system. As far as I know, the .validation was automatically applied for package updates.

Anyways, thanks for correcting me.

**Comment by lychichem** [comment]
The real problem for vanity URL is not automation, it's we cannot make end users update to latest version by running winget  upgrade.

**Comment by Trenly** [comment]
> The real problem for vanity URL is not automation, it's we cannot make end users update to latest version by running winget upgrade.

Winget upgrade would still work, as long as the package is kept up to date and the version changes with every build?

**Comment by lychichem** [comment]
> > The real problem for vanity URL is not automation, it's we cannot make end users update to latest version by running winget upgrade.
> 
> Winget upgrade would still work, as long as the package is kept up to date and the version changes with every build?

No, because any old version will be deleted by wingetbot, so users with older versions installed cannot get any update info by running `winget upgrade`.

**Comment by Trenly** [comment]
> > > The real problem for vanity URL is not automation, it's we cannot make end users update to latest version by running winget upgrade.
> > 
> > 
> > Winget upgrade would still work, as long as the package is kept up to date and the version changes with every build?
> 
> No, because any old version will be deleted by wingetbot in this situation, so users with older versions installed cannot get any update info by running `winget upgrade`. I used to do same thing to VClibs, and got this reply from one of the moderators.

That isn't how winget works though; Winget checks the control panel for currently installed versions and compares against the versions available. You can see that it works fine for many packages like the GeoGebra packages. The reason that no update info would be found in some specific packages where the DisplayVersion key isn't set by the installer, or where the version is the same.

I think that we have to be careful not to generalize, since each package is unique in its own way

**Comment by vedantmgoyal2009** [comment]
@Trenly, It may NOT be a good idea but still, I think the packages which have vanity URLs and are ignored by @wingetbot, we can update them through a GitHub Action.

**Comment by Trenly** [comment]
cc @jedieaston, @OfficialEsco, and @ItzLevvie for their inputs here 
> @Trenly, It may NOT be a good idea but still, I think the packages which have vanity URLs and are ignored by @wingetbot, we can update them through a GitHub Action.

Thats not my concern here. The concern isn't that it can't be updated or can't be automated. The concern is that because it is a nightly build, by the time automation catches the new build, a PR is made, the build passes validation, is tested and approved by moderators, and then gets merged in, there won't be any availability of the package. By using a vanity URL for a nightly package, it will likely result in the package being unusable ~50% of the time due to Hash Mismatch

**Comment by OfficialEsco** [comment]
> > > > The real problem for vanity URL is not automation, it's we cannot make end users update to latest version by running winget upgrade.
> > > 
> > > 
> > > Winget upgrade would still work, as long as the package is kept up to date and the version changes with every build?
> > 
> > 
> > No, because any old version will be deleted by wingetbot in this situation, so users with older versions installed cannot get any update info by running `winget upgrade`. I used to do same thing to VClibs, and got this reply from one of the moderators.
> 
> That isn't how winget works though; Winget checks the control panel for currently installed versions and compares against the versions available. You can see that it works fine for many packages like the GeoGebra packages. The reason that no update info would be found in some specific packages where the DisplayVersion key isn't set by the installer, or where the version is the same.
> 
> I think that we have to be careful not to generalize, since each package is unique in its own way

We can use Google Chrome as a example
```
Publisher: Google LLC
PackageName: Google Chrome
PackageVersion: 96.0.4664.110
```
WinGet searches for `Google LLC *Wildcard*` + `Google Chrome *Wildcard*` which have the best match against the PackageID `Google.Chrome`, if the current version field is lower than `96.0.4664.110` it will show as a Update.

Google Chrome does have proper branches like Beta, Dev, Canary with different PackageNames, so not to make those branches match against the Stable version we create Packages for that
```
Publisher: Google LLC
PackageName: Google Chrome Beta
```

If Google Chrome's Beta branch showed up as `Google Chrome` we would have to use the PackageName `Google Chrome (Beta)` as a Internal PackageName currently there is nothing we can do about that, in the future Channels can deal with that.

However Google Chrome is a MSI which use a different `ProductCode` for each installer, so if the `Publisher` + `PackageName` does not match against anything it would use the `ProductCode` for the matching instead of the `Publisher` + `PackageName`.

NSIS/Inno does have a ProductCode, but that is something the Publisher set permanently (Sorta like the PackageFamilyName for Microsoft Store pacakges)
These can look like 
* GitHubDesktop
* GitHubDesktop_is1
* 771FD6B0-FA20-440A-A002-3B3BAC16DC50
* {771FD6B0-FA20-440A-A002-3B3BAC16DC50}
* {771FD6B0-FA20-440A-A002-3B3BAC16DC50}_is1

> @Trenly, It may NOT be a good idea but still, I think the packages which have vanity URLs and are ignored by @wingetbot, we can update them through a GitHub Action.

The Vanity Urls that are ignored by WinGet usually have some problems like Returning 404 in YamlCreate, but wingetbot also runs EVERY url in the Repo multiple times so its probably blacklisted and blocked for too many connections etc

> Thats not my concern here. The concern isn't that it can't be updated or can't be automated. The concern is that because it is a nightly build, by the time automation catches the new build, a PR is made, the build passes validation, is tested and approved by moderators, and then gets merged in, there won't be any availability of the package. By using a vanity URL for a nightly package, it will likely result in the package being unusable ~50% of the time due to Hash Mismatch

This is why i haven't bothered with Nightly's, its just too much, and 99% of the time these softwares auto update, none manually installs a software daily to keep it updated..

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
This comment is response to "http://github.com/microsoft/winget-pkgs/issues/38444#issuecomment-999707624".

I am solely utilising "nightly" versions of software if they are existent for the product that I am desiring utilisation of.
